src/utils/billing-utils.ts

```diff
@@ -116,30 +118,34 @@ export const formatBillingCatalogStats = (
             const totalCost = evaluateTotalCost(dataVolume, taskCount);
 
             if (billingRecordIndex === -1) {
```

Comment:
This block is pushing how far we can go with just duplicating stuff to save time. That object is pretty big and contains logic. This eventually needs to be shared.